src/include/distributed/utils/array_type.h

```diff
@@ -0,0 +1,23 @@
+/*-------------------------------------------------------------------------
+ *
+ * array_type.h
+ *	  Utility functions for dealing with array types.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef CITUS_ARRAY_TYPE_H
+#define CITUS_ARRAY_TYPE_H
+
+#include "postgres.h"
+
+#include "utils/array.h"
+
+
+extern Datum * DeconstructArrayObject(ArrayType *arrayObject);
+extern int32 ArrayObjectCount(ArrayType *arrayObject);
```

Comment:
minor: maybe move `DatumArrayToArrayType` function to here as well 
```
src/include//distributed/listutils.h:extern ArrayType * DatumArrayToArrayType(Datum *datumArray, int datumCount,

```

Reply:
moved